main.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"archive/zip"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/input"
+)
+
+const (
+	targetExtension string = ".zip"
+)
+
+// ConfigsModel ...
+type ConfigsModel struct {
+	SourcePath string
+	TargetPath string
+}
+
+func main() {
+	configs := createConfigsModelFromEnvs()
+
+	fmt.Println()
+	configs.print()
+
+	if err := configs.validate(); err != nil {
+		fail("Issue with input: %s", err)
+	}
+
+	_, err := compress(configs.SourcePath, configs.TargetPath)
+
+	if err != nil {
+		fail("Issue with compress: %s", err)
+		return
+	}
+}
+
+// Adds the file extension to the end of the targetpath if it's needed.
+func fixTargetExtension(targetPath string) string {
+	if strings.HasSuffix(targetPath, targetExtension) {
+		return targetPath
+	}
+	return fmt.Sprintf("%s%s", targetPath, targetExtension)
+}
+
+func compress(sourcePath string, targetPath string) (string, error) {
+	targetPath = fixTargetExtension(targetPath)
+
+	validateTargetPath(targetPath)
+
+	zipfile, err := os.Create(targetPath)
+
+	if err != nil {
+		return "", err
+	}
+
+	defer zipfile.Close()
+
+	archive := zip.NewWriter(zipfile)
+	defer archive.Close()
+
+	info, err := os.Lstat(sourcePath)
+	if err != nil {
+		return "", err
+	}
+
+	var baseDir string
+	if !info.IsDir() {
+		baseDir = filepath.Base(sourcePath)
+	}
+
+	filepath.Walk(sourcePath, func(path string, info os.FileInfo, err error) error {
```

Comment:
void using `path` as variable nem, there is a standard library package: `path`, use: `pth` instead